radix-engine/src/blueprints/pool/one_resource_pool/blueprint.rs

```diff
@@ -363,13 +362,15 @@ impl OneResourcePoolBlueprint {
             reserves > Decimal::ZERO,
         ) {
             (false, false) => Ok(amount_of_contributed_resources),
-            (false, true) => Ok(amount_of_contributed_resources.safe_add(reserves).unwrap()),
+            (false, true) => Ok(amount_of_contributed_resources
+                .safe_add(reserves)
+                .ok_or_else(|| OneResourcePoolError::DecimalOverflowError)?),
             (true, false) => Err(OneResourcePoolError::NonZeroPoolUnitSupplyButZeroReserves),
             (true, true) => Ok(amount_of_contributed_resources
                 .safe_mul(pool_unit_total_supply)
```

Comment:
We should be doing the division first here and then the multiplication so that we do not overflow.